lynx/common/actions/drop.py

```diff
@@ -0,0 +1,48 @@
+from dataclasses import dataclass
+from collections import defaultdict
+
+from lynx.common.actions.action import Action
+from lynx.common.actions.common_requirements import CommonRequirements
+from lynx.common.actions.create_object import CreateObject
+from lynx.common.actions.update_resources import UpdateResources
+from lynx.common.object import Object
+from lynx.common.vector import Vector
+
+
+@dataclass
+class Drop(Action):
+    """
+    Action used to empty inventory of agent.
+    If target_position is equal to drop_area_position, we should increase global points and resources.
+    If target_position is different than drop_area_position then we should create each object from inventory on choosen
+    square (target_position)
+    """
+    object_id: int = -1
+    target_position: Vector = Vector(0, 1)
+
+    def apply(self, scene: 'Scene') -> None:
+        agent: Object = scene.get_object_by_id(self.object_id)
+        player_name = agent.owner
+        if self.target_position == scene.get_drop_area_of_a_player(player_name):
```

Comment:
Definitely create helper function for these 2 cases e.g. `drop_in_drop_area`, `drop_in_overworld`